src/DemuxTopic.h

```diff
@@ -30,10 +31,12 @@ class DemuxTopic : public TimeDifferenceFromMessage, public MessageProcessor {
   ProcessMessageResult process_message(char *msg_data, int msg_size);
   /// Implements TimeDifferenceFromMessage.
   DT time_difference_from_message(char *msg_data, int msg_size);
-  std::vector<Source> &sources();
-  template <typename... Args> Source &add_source(Args &&... args) {
-    _sources.emplace_back(std::forward<Args>(args)...);
-    return _sources.back();
+  std::unordered_map<std::string, Source> &sources();
+  Source &add_source(Source &&source) {
+    using std::move;
```

Comment:
This is a very minor point but I think using `using` here just adds a line of code for no readability improvement.